index/src/executors/bridge_executors/moonbeam_to_phala.rs

```diff
@@ -0,0 +1,100 @@
+use crate::prelude::BridgeExecutor;
+use crate::prelude::Error;
+use crate::transactors::XtokenClient;
+use alloc::vec::Vec;
+use pink_web3::ethabi::Address;
+
+use pink_web3::{
+    api::{Eth, Namespace},
+    contract::Contract,
+    keys::pink::KeyPair,
+    transports::PinkHttp,
+};
+
+#[derive(Clone)]
+pub struct Moonbeam2PhalaExecutor {
+    bridge_contract: XtokenClient,
+}
+
+impl Moonbeam2PhalaExecutor {
+    #[allow(dead_code)]
+    pub fn new(rpc: &str, bridge_address: Address) -> Self {
+        let eth = Eth::new(PinkHttp::new(rpc));
+        let bridge_contract = XtokenClient {
+            contract: Contract::from_json(
+                eth,
+                bridge_address,
+                include_bytes!("../../abis/xtokens-abi.json"),
+            )
+            .expect("Bad abi data"),
+        };
+
+        Self { bridge_contract }
+    }
+}
+
+impl BridgeExecutor for Moonbeam2PhalaExecutor {
+    fn transfer(
+        &self,
+        signer: [u8; 32],
+        asset_contract_address: Vec<u8>,
+        recipient: Vec<u8>,
+        amount: u128,
+    ) -> core::result::Result<(), Error> {
+        let signer = KeyPair::from(signer);
+        let token_address = Address::from_slice(&asset_contract_address);
+        _ = self
+            .bridge_contract
+            .transfer(
+                signer,
+                token_address,
+                amount,
+                // parents = 1
+                1,
+                // parachain
+                2035,
```

Comment:
Do you consider adding `parachain id` as a variant of struct `Moonbeam2PhalaExecutor`, Looks like this implementation can be reused in Moonriver > Khala bridge, only need to change `xtoken contract address` and `khala parachian id` (2004)

Reply:
it's done, check this out: https://github.com/tenheadedlion/index-contract/blob/m2p/index/src/transactors/xtokens.rs#L24

the reason why 2035 is here is that the executor's destination chain is Phala